src/SprykerEco/Zed/Payone/Persistence/PayoneRepository.php

```diff
@@ -42,6 +56,112 @@ public function findPaymentPayoneOrderItemByFilter(PayoneOrderItemFilterTransfer
         return $paymentPayoneOrderItemTransfers;
     }
 
+    /**
+     * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
+     *
+     * @return \Generated\Shared\Transfer\PayonePaymentTransfer
+     */
+    public function getPayonePaymentByOrder(OrderTransfer $orderTransfer): PayonePaymentTransfer
+    {
+        $payment = $this->createPaymentByOrderId($orderTransfer->getIdSalesOrder())->findOne();
+        $paymentDetail = $payment->getSpyPaymentPayoneDetail();
```

Comment:
Extract mapping to a mapper `PayonePersistenceMapper::mapPaymentPayoneEntityToPayonePaymentTransfer(paymentPayoneEntity, new PayonePaymentTransfer())`.

Reply:
++